src/utils/entitySpec.js

```diff
@@ -15,6 +15,26 @@ const toLowerCaseMap = {
   mavencentralsource: NONE
 }
 
+const NPM_WEBSITE = 'npmjs.com'
```

Comment:
Rather than a destructured set of fields to capture each provider's details, we should have a first class object to represent a provider. That would have the path, as well as the function that teases apart the dropped URL and forms coordinates. Then we can have provider support plugged in by just discovering all the provider objects. There is more work to do other places to make that a reality but lets start down that path here.

Reply:
Just to be sure to have understood well the suggestion:
- a Provider class should be implemented
- a specific Provider (like NpmProvider, and so on) should be implemented based on the generic Provider object, exposing specific methods and values
- each provider should contain specific rules on how to recognize a URL (like .com or .org extension for npmjs)
- a function that parses the URL and recognize which provider object is related

Is that right?

Reply:
More or less. Just imagine that a month from now someone wants to add support for some new provider. Do they roam about the code adding a bit here, a bit there, tweak and if or switch statement, ... of do they implement a new class with a bit of data and a handful of methods? I'm hoping for the latter.

Reply:
The idea i have is to build a generic Provider class, with shared structure and methods.
A specific provider could be an extension of the generic one, this would require to the developer to only build specific methods

Reply:
sure. lets not over engineer but at least some structure here would be good

Reply:
I will provide a refactor of this code while working on #168 , which actually adds a new type of supported URL

Reply:
@jeffmcaffer would you take a look at https://github.com/WebYourMind/website/pull/51/files#diff-b83a5a3eca3acc3a9eebddb8273448f3

I've implemented a solution proposal to address the Provider structure.

As i mentioned before, the idea is to allow the user to implement new specific providers, which are actually an extension of a generic one.

The Provider class should contain all the methods usable on each single specific provider.